package/environments/development.js

```diff
@@ -18,35 +18,41 @@ if (runningWebpackDevServer) {
     })
   }
 
-  devConfig = merge(devConfig, {
-    devServer: {
-      clientLogLevel: 'none',
-      compress: devServer.compress,
-      quiet: devServer.quiet,
-      disableHostCheck: devServer.disable_host_check,
-      host: devServer.host,
-      port: devServer.port,
-      https: devServer.https,
-      hot: devServer.hmr,
-      contentBase,
-      inline: devServer.inline || devServer.hmr,
-      injectClient: devServer.hmr,
-      injectHot: devServer.hmr,
-      useLocalIp: devServer.use_local_ip,
-      public: devServer.public,
-      publicPath,
-      historyApiFallback: { disableDotRule: true },
-      headers: devServer.headers,
-      overlay: devServer.overlay,
-      stats: {
-        colors: true,
-        entrypoints: false,
-        errorDetails: true,
-        modules: false,
-        moduleTrace: false
-      },
-      watchOptions: devServer.watch_options
+  const devServerConfig = {
+    devMiddleware: {
+      publicPath
+    },
+    compress: devServer.compress,
+    allowedHosts: devServer.allowed_hosts,
+    host: devServer.host,
+    port: devServer.port,
+    https: devServer.https,
+    hot: devServer.hmr,
+    liveReload: !devServer.hmr,
+    historyApiFallback: { disableDotRule: true },
+    headers: devServer.headers,
+    static: {
+      publicPath: contentBase
     }
+  }
+
+  if (devServer.static) {
+    devServerConfig.static = { ...devServerConfig.static, ...devServer.static }
+  }
+
+  if (devServer.client) {
+    devServerConfig.client = devServer.client
+  }
+
+  devConfig = merge(devConfig, {
```

Comment:
Doesn't this override all local (user-side) changes to the config though? We're trying to change this config and our changes keep getting overridden by defaults.

Reply:
What specific changes are you trying to make that are being overridden? I was under the impression that `merge` would more or less take the existing `devConfig` which is from the `yaml` file and stick on the `stats` and `devServer` values.

From what I can see, this is a similar pattern to webpacker 4 (old line 21 of this diff)